src/WebJobs.Script/Description/PowerShell/PowerShellFunctionInvoker.cs

```diff
@@ -87,7 +87,9 @@ public class PowerShellFunctionInvoker : ScriptFunctionInvokerBase
                         System.Management.Automation.PowerShell.Create())
                 {
                     powerShellInstance.Runspace = runspace;
-                    _moduleFiles = GetModuleFilePaths(_host.ScriptConfig.RootScriptPath, _functionName);
+                    _moduleFiles = FindDuplicateModules(GetModuleFilePaths(_host.ScriptConfig.RootScriptPath, _functionName), _host.ScriptConfig.RootScriptPath, _functionName);
+                    // Remove duplicates from _moduleFiles and only keep the ones closest to the Function in moduleDirectory
+
                     if (_moduleFiles.Any())
                     {
                         powerShellInstance.AddCommand("Import-Module").AddArgument(_moduleFiles);
```

Comment:
Well, I'd prefer to not have any auto-import, at all, not even on a function level.
I was told that auto-import should stay so to not break current behaviour.
I would either opt for removing auto-import everywhere or add my approach here and auto-import.
How does it work in other languages on Functions?
Having implicit imports in one case and explicit in other cases makes it quite confusing, unless we start working with PSModulePath.
PowerShell users will either expect auto loading to just work, wherever they put stuff or not.

One option would be to add the two "modules" locations to `PSModulePath` and use PowerShell's auto-loading of modules, but this again would have performance implications.
In the end, PowerShell on Functions will never perform as well as the other languages.